### PR TITLE
Fix/windows build failing

### PR DESCRIPTION
**Pull request by jeremyfelder**
## Describe the changes

This PR fixes the rust build for windows by:
- Making the curve config's ntt size conditionally accessed when creating fields
- Removing the extern C function that returned a UDT containing non-POD types and replacing it with a `default_config` function on the Rust bindings side

### CODE REVIEW

File: icicle/appUtils/ntt/ntt.cu [hunk 1]
```diff
@@ -386,7 +386,8 @@ namespace ntt {
     // but it's a singleton that is supposed to be initialized once per program lifetime
     if (!Domain<S>::twiddles) {
       S omega = primitive_root;
-      for (int i = 0; i < S::TWO_ADICITY; i++)
+      unsigned omegas_count = S::get_omegas_count();
```
[the remark this answers is not in the document]
Comment (jeremyfelder) [reply icicle/appUtils/ntt/ntt.cu:389]: This solves the first issue of there not being a `omegas_count` member on an `fq_config`

File: icicle/primitives/field.cuh [hunk 2]
```diff
@@ -85,6 +84,26 @@ public:
...
+      return CONFIG::omegas_count;
+    } else {
+      return 0;
+    }
+  }
+
+  template <typename T>
+  static constexpr bool has_member_name()
+  {
+    // if (std::is_same<unsigned, decltype(T::omegas_count)>) {
```
Comment (yshekel) [inline icicle/primitives/field.cuh:99]: please remove commented out code

File: icicle/primitives/field.cuh [hunk 3]
```diff
@@ -85,6 +84,26 @@ public:
...
+    if (sizeof(T::omegas_count) > 0) {
```
Comment (yshekel) [inline icicle/primitives/field.cuh:100]: minor comment but I would rewrite as 'return sizeof(T::omegas_count) > 0' and avoid the if-else.

**Review by yshekel — APPROVED** [review]
Looks good to me. I wrote minor comments.

**Pull request merged**